test/parallel/test-buffer.js

```diff
@@ -28,6 +28,9 @@ var c = new Buffer(512);
 console.log('c.length == %d', c.length);
 assert.strictEqual(512, c.length);
 
+var d = new Buffer([]);
+assert.strictEqual(0, d.length);
+
```

Comment:
Can you also add the following:
```js
var ui32 = new Uint32Array(4).fill(42);
var b = Buffer(ui32);
assert.deepEqual(ui32, b);
```
This is a test that should have been added previously, so is not technically part of this PR, but might as well add it since we're here.